### PR TITLE
core: deprecate DEVICEx_TYPE/_PA_BASE/_SIZE

### PR DESCRIPTION
**Pull request by etienne-lms**
This is a minor change, rather a cleanup for a deprecated resource since OP-TEE 2.1.0.

Macros `DEVICEx_TYPE,` `DEVICEx_PA_BASE` and `DEVICEx__SIZE` used to help platform to register their address range mapping requirements. These are now deprecated since platform should use the more flexible `register_phys_mem()` macro.

This change removes all occurrences of `DEVICEx_TYPE/_PA_BASE/_SIZE` and use the `register_phys_mem()` instead in related platform specific implementation

### CODE REVIEW

**Comment by jenswi-linaro** [comment]
LGTM, please fix the travis errors.

File: core/arch/arm/include/mm/core_mmu.h [hunk 1]
```diff
@@ -111,11 +111,14 @@ struct core_mmu_phys_mem {
 	size_t size;
 };
 
-#define register_phys_mem(type, addr, size) \
-	static const struct core_mmu_phys_mem __phys_mem_ ## addr \
+#define __phys_mem2(type, addr, size, id) \
```
Comment (jenswi-linaro) [inline core/arch/arm/include/mm/core_mmu.h]: Not that it matters much, but these two new `__phys_mem1` and `__phys_mem2` could have been made to cover what `__register_sdp_mem1` and `__register_sdp_mem2` is doing too.

**Comment by jenswi-linaro** [comment]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by etienne-lms** [comment]
Comments address. I have reordered the change into 2 commits.
Jens, please review again( r-b not yet applied), but let's wait after the release before we merge this.

File: core/arch/arm/plat-mediatek/main.c [hunk 2]
```diff
@@ -38,6 +38,14 @@
 
 static void main_fiq(void);
 
+register_phys_mem(MEM_AREA_IO_NSEC,
+		  ROUNDDOWN(CONSOLE_UART_BASE, CORE_MMU_DEVICE_SIZE),
```
Comment (jenswi-linaro) [inline core/arch/arm/plat-mediatek/main.c]: The `ROUNDDOWN()` isn't needed, that's taken care of when the address ranges are collected.

Comment (etienne-lms) [inline core/arch/arm/plat-mediatek/main.c]: that's right, and i don't really like that. 
Some platform may need to use fine mapping for iomem, and one must know that `MEM_AREA_IO_SEC/_NSEC` will widely map the address range.
Whatever, this shall be treated later.

Regarding current change, I did not want this change to modify the implementation related to code I do not own. However, considering the current implementation of the core, I agree with you. As Linaro is the MTK8713 maintainer and your asking, ok remove these rounding.

Comment (etienne-lms) [inline core/arch/arm/plat-mediatek/main.c]: By the way, mtk8731 registers the gic iomem but does not use the GIC.
Shouldn't we better remove the GIC mapping ?

Comment (jenswi-linaro) [inline core/arch/arm/plat-mediatek/main.c]: Fair enough.

Comment (jenswi-linaro) [inline core/arch/arm/plat-mediatek/main.c]: Regarding your last comment, I wouldn't mind.

**Comment by jenswi-linaro** [comment]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by etienne-lms** [comment]
rebased, tag applied.

**Comment by jbech-linaro** [comment]
@etienne-lms , I've sneaked in (merged) some of Jens' patches, so would you mind rebase this patch set again and then I'll merge it right away?

**Comment by etienne-lms** [comment]
sure.

**Pull request merged**